### PR TITLE
Add sponsor page & Create Sponsor Us Dev Branch

### PR DESCRIPTION
**Pull request by mike10911**
Establishes a Sponsor Us Page and Dev Branch

Change list:

- Added a new page to the /pages call `sponsor-us.tsx`
- In `sponsor-us.tsx` added filler components
- Added "Sponsor Us" to mainSiteTabine and it path "sponsor-us"
- Created a branch called sponsor-us-dev-branch for devs to merge any sponsor us page PRs into

Tested:

- localhost

Screenshots & Screencasts:

### New Page
Screenshot 2023-07-08 at 7 03 41 PM

### Nav Bar w/ "Sponsor Us"
Screenshot 2023-07-08 at 7 07 11 PM

### New Sponsor Us Dev Branch
Screenshot 2023-07-08 at 7 06 43 PM

### CODE REVIEW

File: main-site/src/pages/sponsor-us.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    if (isDay && !isDesktop) {
+      return 'mobile-light-background';
+    }
+    if (!isDay && isDesktop) {
+      return 'dark-background-desktop';
+    }
+    return 'mobile-dark-background';
+  };
+  useEffect(() => {
```
Comment (surbhi-gulati) [inline main-site/src/pages/sponsor-us.tsx:33]: nit: newline

File: main-site/src/pages/sponsor-us.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import type { HeadFC } from 'gatsby';
+import '../../../shared-ui/style/globals.css';
```
Comment (surbhi-gulati) [inline main-site/src/pages/sponsor-us.tsx:3]: line 3 and 4 duplicate import

File: main-site/src/pages/sponsor-us.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import Footer from '../../../shared-ui/components/footer/Footer';
+import useMatchMedia from 'react-use-match-media';
+import { min } from '../../../shared-ui/lib/responsive';
+import { StyledPageContainer } from '../../../shared-ui/styled-components/Background.styles';
+import AboutSection from '../../components/about-section/AboutSection';
+import ExploreSection from '../../components/explore-section/ExploreSection';
+import EventsCalendarSection from '../../components/events-calendar-section/EventsCalendarSection';
+import TestimonialsSection from '../../components/testimonials-section/TestimonialsSection';
+
+const SponsorUsPage: React.FC = () => {
```
Comment (surbhi-gulati) [inline main-site/src/pages/sponsor-us.tsx:15]: these things are fine as-is at the moment but can we make tech debt ticket to clean up reused state things:
- standardize app-wide state of isDay, isDesktop cleanly thru a useContext or something
- standardize the header / footer inclusion using something like <Outlet />

File: main-site/src/pages/sponsor-us.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      <EventsCalendarSection />
+      <TestimonialsSection />
+      <Footer tabs={mainSiteTabInfo} isDay={isDay} />
+    </StyledPageContainer>
+  );
+};
+
+export default SponsorUsPage;
+
+export const SponsorUsTitle: HeadFC = () => <title>Sponsor Us</title>;
```
Comment (surbhi-gulati) [inline main-site/src/pages/sponsor-us.tsx:55]: it doesn't look like this is properly routing to a new page right now? it seems to show the main page contents. unsure if you meant to do this within this PR or a follow-up PR

**Review by surbhi-gulati — CHANGES_REQUESTED** [review]

File: main-site/src/pages/sponsor-us.tsx (hunk 4 above)
Comment (mike10911) [reply main-site/src/pages/sponsor-us.tsx:55]: I just copied in some filler components for now from the main page. It is routing correctly, it just as a few of the same components from the main page.

File: main-site/src/pages/sponsor-us.tsx (hunk 3 above)
Comment (mike10911) [reply main-site/src/pages/sponsor-us.tsx:15]: using something like...?

File: main-site/src/pages/sponsor-us.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    setIsLoading(false);
+  }, []);
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <StyledPageContainer className={getBackgroundClassName()}>
+      <Header tabs={mainSiteTabInfo} isDay={true} />
```
Comment (kzeng24) [inline main-site/src/pages/sponsor-us.tsx]: Looks good to me! By the way, should the isDay property take in the variable isDay instead of true?

**Review by kzeng24 — APPROVED** [review]

File: shared-ui/lib/data.ts [hunk 6]
```diff
@@ -39,6 +39,7 @@ const mainSiteTabInfo: TabInfo[] = [
   { name: 'About', link: '/#about' },
   { name: 'Calendar', link: '/#calendar' },
   { name: 'FAQ', link: '/#faq' },
+  { name: 'Sponsor Us', link: '/sponsor-us' },
```
Comment (kzeng24) [inline shared-ui/lib/data.ts:42]: Also just curious: is the sponsor us page going to be a separate new page? Or is it supposed to be linking to the existing sections on the home page?

File: main-site/src/pages/sponsor-us.tsx (hunk 3 above)
Comment (surbhi-gulati) [reply main-site/src/pages/sponsor-us.tsx:15]: <Outlet /> oops

**Review by surbhi-gulati — APPROVED** [review]

File: shared-ui/lib/data.ts (hunk 6 above)
Comment (mike10911) [reply shared-ui/lib/data.ts:42]: It is a separate page! That's why it doesn't have a # in the path!

**Pull request merged**